pkg/resources/secret_with_basic_authentication_acceptance_test.go

```diff
@@ -0,0 +1,212 @@
+package resources_test
+
+import (
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance/bettertestspoc/assert/resourceshowoutputassert"
+	"testing"
+
+	acc "github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance/bettertestspoc/assert"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance/bettertestspoc/assert/resourceassert"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance/bettertestspoc/config"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance/bettertestspoc/config/model"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance/helpers/random"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance/importchecks"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance/planchecks"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/helpers"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/provider/resources"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	tfjson "github.com/hashicorp/terraform-json"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/plancheck"
+	"github.com/hashicorp/terraform-plugin-testing/tfversion"
+)
+
+func TestAcc_SecretWithBasicAuthentication_BasicFlow(t *testing.T) {
```

Comment:
What will happen if we change the secret type externally (there is no alter AFAIK but... someone can always drop and create with same name but different type). For user resource it his handled by conditional force new, you can check the user_type computed attribute.

We do not have to add this logic in the current change but I think it should be added in one of the next PRs.

cc: @sfc-gh-jcieslak @sfc-gh-jmichalak

Reply:
I will add tests cases for such behavior in next PRs, and probably use the same method as for the user resource

Reply:
@sfc-gh-jmichalak true, I'll add a task to check existing ones and I will add it to the redesign guidelines